### ISSUE TITLE
A question on args validation for builtin functions

### ISSUE THREAD

**Issue by zyguan** [pingcap/tidb#2333]
Hi, I have a question on args validation for builtin functions. It seems that the behavior of tidb (errno) is inconsistent with the mysql, although the error message is clear enough. Does it matter in some cases?

1. What did you do?
```
tidb> SELECT GREATEST(1);
```

2. What did you expect to see?
```
ERROR 1105 (HY000): number of function arguments must in [2, -1].
```

3. What did you see instead?
```
ERROR 1582 (42000): Incorrect parameter count in the call to native function 'GREATEST'
```

4. What version of Go are you using (`go version`)?
```
go version go1.7.4 linux/amd64
```

**Comment by shenli** [comment]
@zyguan Thanks for your feedback!
The error message means the number of arguments is not right. For greatest(), it need at least 2 argument. 
"[2, -1]" means the function need at least 2 arguments and no upper limit.

**Comment by shenli** [comment]
@zyguan This will be fixed in https://github.com/pingcap/tidb/pull/2335

**Comment by zyguan** [comment]
Thanks @shenli ! Is it a urgent issue? I may be able to fix #2335.

**Comment by zyguan** [comment]
Sorry, It's been fixed by #2335, forget my previous comment.
